pilot/pkg/xds/ads.go

```diff
@@ -235,6 +240,28 @@ func (s *DiscoveryServer) StreamAggregatedResources(stream discovery.AggregatedD
 	reqChannel := make(chan *discovery.DiscoveryRequest, 1)
 	go s.receive(con, reqChannel, &receiveError)
 
+	pushErrChannel := make(chan error, 1)
+
+	// handle active xds push (resources or configs update)
+	go func() {
+		select {
+		case pushEv := <-con.pushChannel:
```

Comment:
I think this introduces another race condition. We now have two readers on `con.pushChannel`, go may select either one arbitrarily.

Are you sure this is actually a bug? I think the sequence would be:

CDS REQ
Push comes in from config X: We push the new CDS, but EDS, LDS, and RDS are not watched so we do not update those
EDS, LDS, RDS requests come in. We send back the newest config, including config X

Reply:
Oh, it was refactored by adding `watchedResources`. looks good